spylon_kernel/scala_magic.py

```diff
@@ -78,28 +87,70 @@ def _loop_alive(self):
             yield gen.sleep(0.01)
 
     def line_scala(self, *args):
-        """
-        %scala CODE - evaluate code as Scala
-        This line magic will evaluate the CODE (either expression or
-        statement) as Scala code.
-
-        Examples:
-            %scala val x = 42
-            %scala import scala.math
-            %scala x + math.pi
+        """%scala CODE - evaluates a line of code as Scala
+
+        Parameters
+        ----------
+        *args : list of string
+            Line magic arguments joined into a single-space separated string
+
+        Examples
+        --------
+        %scala val x = 42
+        %scala import scala.math
+        %scala x + math.pi
         """
         code = " ".join(args)
         self.eval(code, True)
 
+    @option(
+        "-e", "--eval_output", action="store_true", default=False,
+        help="Evaluate the return value from the Scala cell as code in the kernel language."
+    )
```

Comment:
Magics are sort of like command lines. They can have arguments like a command line. `%%script` is a good example in IPython. I'll note why argparse is being used.